client/src/modules/dashboard/index.jsx

```diff
@@ -1,77 +1,226 @@
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 import _map from 'lodash/map';
+import classNames from 'classnames';
 
+import { SvgIcon } from 'assets/images/icons';
+import PlusIcon from 'assets/images/plus-solid.svg';
+import CreationForm from 'common/components/CreationForm';
 import {
-  ClipboardSolid as ShoppingListIcon,
-  UsersSolid as CohortIcon
-} from 'assets/images/icons';
-import { fetchShoppingListsMetaData } from 'modules/shopping-list/model/actions';
-import { fetchCohorts } from 'modules/cohort/model/actions';
+  createShoppingList,
+  fetchShoppingListsMetaData
+} from 'modules/shopping-list/model/actions';
+import { createCohort, fetchCohorts } from 'modules/cohort/model/actions';
 import { getShoppingLists } from 'modules/shopping-list/model/selectors';
 import { getCohorts } from 'modules/cohort/model/selectors';
+import { getCurrentUser } from 'modules/authorization/model/selectors';
+import Toolbar from 'common/components/Toolbar/Toolbar';
+import { UserPropType } from 'common/constants/propTypes';
+import ToolbarItem from 'common/components/Toolbar/components/ToolbarItem/ToolbarItem';
 import CardItem from './CardItem';
+import { IconType } from 'common/constants/enums';
 
 class Dashboard extends Component {
+  constructor(props) {
+    super(props);
+    this.cohortFormRef = React.createRef();
+    this.listFormRef = React.createRef();
```

Comment:
try not use refs - from docs: "Avoid using refs for anything that can be done declaratively."